src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts

```diff
@@ -0,0 +1,232 @@
+		private _enableNative = false;
+		private _isOpen = false;
+		private _searchGlassButton: HTMLElement;
+		private _searchInput: HTMLInputElement;
+		/**
+		 * Creates an instance of Search.
+		 *
+		 * @param {string} uniqueId
+		 * @param {JSON} configs
+		 * @memberof Search
+		 */
+		constructor(uniqueId: string, configs: JSON) {
+			super(uniqueId, new SearchConfig(configs));
+		}
+
+		private _nativeSearchBehavior(): void {
+			// Store the sate of native behavior
+			this._enableNative = true;
+
+			// Add class for native behavior
+			this.selfElement.classList.add(Enum.CssClass.PatternNative);
+
+			// Define the input selector
+			this._searchInput = Helper.Dom.TagSelector(
+				this.selfElement,
+				OSFramework.OSUI.GlobalEnum.HTMLElement.Input
+			) as HTMLInputElement;
+
+			// Create HTML element for native behavior
+			this._searchGlassButton = document.createElement(OSFramework.OSUI.GlobalEnum.HTMLElement.Div);
+			this._searchGlassButton.classList.add(Enum.CssClass.PatternGlassButton);
+			this.selfElement.appendChild(this._searchGlassButton);
+
+			// Set A11Y properties to elements
+			Helper.A11Y.TabIndexFalse(this._searchInput);
+			Helper.A11Y.TabIndexTrue(this._searchGlassButton);
+			Helper.A11Y.AriaExpandedFalse(this._searchGlassButton);
+			Helper.A11Y.AriaLabel(this._searchGlassButton, 'Open Search');
+
+			// Create span element to create the custom glass on search
+			const glassCircle = document.createElement(OSFramework.OSUI.GlobalEnum.HTMLElement.Span);
+			const glassBar = document.createElement(OSFramework.OSUI.GlobalEnum.HTMLElement.Span);
+
+			// Add classes to each span created
+			glassCircle.classList.add(Enum.CssClass.PatternGlassCircle);
+			glassBar.classList.add(Enum.CssClass.PatternGlassBar);
+
+			// Add elements into Search glass button
+			this._searchGlassButton.appendChild(glassCircle);
+			this._searchGlassButton.appendChild(glassBar);
+
+			// Add event listener to created element
+			this._searchGlassButton.addEventListener(GlobalEnum.HTMLEvent.Click, this._toggleNativeSearch.bind(this));
+		}
+
+		/**
+		 * Method to remove all the assigned events
+		 *
+		 * @private
+		 * @memberof Search
+		 */
+		private _removeEvents(): void {
+			this._searchGlassButton.removeEventListener(GlobalEnum.HTMLEvent.Click, this._toggleNativeSearch);
+		}
+
+		/**
+		 * Method to toogle search visibility in native
+		 *
+		 * @private
+		 * @memberof Search
+		 */
+		private _toggleNativeSearch(): void {
+			if (this._isOpen) {
+				// Remove the isOpen class and update the state variable
+				this.selfElement.classList.remove(Enum.CssClass.PatternIsOpen);
+
+				// Set A11Y properties
+				Helper.A11Y.TabIndexFalse(this._searchInput);
+				Helper.A11Y.AriaExpandedFalse(this._searchGlassButton);
+				Helper.A11Y.AriaLabel(this._searchGlassButton, 'Open Search');
+
+				this._isOpen = false;
+			} else {
+				// Focus on search input if is empty
+				if (this._searchInput && this._searchInput.value === '') {
```

Comment:
Why move the focus only if the input is empty?

Reply:
This is a old use case of the search component and I kept the same on this new one. Basically they can search and keep the value on the input and navigate between data inside the screen.